ACPs/176-dynamic-evm-gas-limit-and-price-discovery-updates/README.md

```diff
@@ -0,0 +1,158 @@
+
+## Specification
+
+### Gas Price Determination
+
+The mechanism to determine the base fee of a block is the same as the one used in ACP-103 to determine the gas price of a block on the P-Chain. This mechanism calculates the gas price for a given block $b$ based on the following parameters:
+
+<div align="center">
+
+| | |
+|---|---|
+| $T$ | the target gas consumed per second |
+| $M$ | minimum gas price |
+| $K$ | gas price update constant |
+| $C$ | maximum gas capacity |
+| $R$ | gas capacity added per second |
+
+</div>
+
+The full specification of this mechanism can be found [here](../103-dynamic-fees/README.md).
+
+### Making $T$ Dynamic
+
+As noted above, the gas price determination mechanism relies on a target gas consumption per second, $T$, in order to calculate the gas price for a given block. $T$ will be adjusted dynamically according to the following specification.
+
+Let $q$ be a non-negative integer that is initialized to 0 upon activation of this mechanism. Let the target gas consumption per second be expressed as:
+
+$$T = P \cdot e^{q/D}$$
+
+where $P$ is the global minimum allowed target gas consumption rate for the network, and $D$ is a constant that helps control the rate of change of the target gas consumption.
+
+After the execution of transactions in block $b$, the value of $q$ can be increased or decreased up to $Q$. It must be the case that $q_{b-1} - Q \leq q_{b} \leq q_{b-1} + Q$, or block $b$ is considered invalid. The amount by which $q$ changes after executing block $b$ is specified by the block builder.
+
+Block builders (i.e. validators), may set their desired value for $T$ (i.e. their desired gas consumption rate) in their configuration, and their desired value for $q$ can then be calculated as:
+
+$$q_{desired} = D \cdot ln(\frac{T}{P})$$
+
+Note that since $q_{desired}$ is only used locally and can be different for each node, it is safe for implementations to approximate the value of $ln(\frac{T}{P})$, and round the resulting value to the nearest integer.
+
+When building a block, builders can calculate their next preferred value for $q$ according to:
+
+  ```python
+  # Calculates a node's new desired value for q given for a given block
+  def calc_next_q(q_current: int, q_desired: int, max_change: int) -> int:
+    if q_desired > q_current:
+        return q_current + min(q_desired - q_current, max_change)
+    else:
+        return q_current - min(q_current - q_desired, max_change)
+  ```
+
+As $q$ is updated after the execution of transactions within the block, $T$ is also updated such that $T = P \cdot e^{q/D}$ at all times. As the value of $T$ adjusts, the value of $R$ (capacity added per second) is also updated such that:
+
+$$R = T \cdot 2$$
+
+This ensures that the gas price remains equally reactive in either direction depending on how far the actual gas consumption rate is from the target, whether it is above or below.
```

Comment:
```suggestion
This ensures that the gas price can increase and decrease at the same rate.
```

Specifically, the rate of gas consumption is in `[0, C]`. Because $C = 2T$, then excess will be increasing at a rate of `T` when the gas consumption rate is `C` and decreasing at a rate of `T` If the gas consumption rate is `0`.